test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py

```diff
@@ -21,14 +21,20 @@
 from test.test_utils import get_framework_and_version_from_tag, get_cuda_version_from_tag
 from packaging.version import Version
 from packaging.specifiers import SpecifierSet
-
 from ...integration import (data_dir, dist_operations_path, fastai_path, mnist_script,
                               DEFAULT_TIMEOUT, mnist_path)
 from ...integration.sagemaker.timeout import timeout
 
 MULTI_GPU_INSTANCE = 'ml.p3.8xlarge'
 
 
+def validate_or_skip_smmodelparallel(ecr_image):
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    image_cuda_version = get_cuda_version_from_tag(ecr_image)
+    if not(Version(image_framework_version) in SpecifierSet(">=1.6,<1.9")) or image_cuda_version != "cu110":
```

Comment:
Thanks for adding this as a function to reduce code duplication in this script!

Please remove the upper-limit on PT version. If the test fails because smmodelparallel doesn't work on a new framework version, we'd like to make a conscious decision to skip testing on that version rather than have the choice be made automatically.

Reply:
Makes sense, will do